proposals/memory-control/Overview.md

```diff
@@ -1,38 +1,81 @@
+- **Allowing applications to release memory to the host.** WebAssembly memory can only grow and never shrink, and there are no further APIs for releasing memory to the host. This means that a WebAssembly application's memory footprint can only increase, which impacts system performance and can on some devices (especially mobile devices) result in the application being terminated. (Example: [Unity](https://github.com/WebAssembly/design/issues/1397))
+- **Supporting memory protection.** There are many use cases for making memory inaccessible or read-only. For example, toolchains may wish to make constant data read-only to prevent corruption, media codecs may wish to prevent clients from manipulating buffers that the decoder is actively using, and languages may wish to trap on null pointer accesses.
 
-The proposal is currently scoped to work with a single linear memory, i.e. all of the proposed operations will operate on a single linear memory, and the operations proposed here will only be available as JS API functions. The core spec should have some notion of a “special” mappable memory, but outside of that we will likely not expose any other operations unless there is a consistent way that we can expose them that will be useful to other embeddings. Earlier versions of this proposal exploration included statically declared memories, with bind/unbind APIs, and exploring support for first class memories. Some reasons for scoping the proposal to use a single linear memory are: 
+All of these concerns can be addressed with a set of features based on system virtual memory APIs.
 
-- Ease of portability: Non-trivial addition of work when porting existing applications. A big win for exisiting applications is that porting to Wasm should in most cases work pretty much out of the box, having separate memories will need additional annotations, and only address a smaller set of application use cases
-- Better toolchain integration: While LLVM supports multiple address spaces, most C/C++, Rust programs assume a single linear memory space, and that all pointers are accessible.
-- Graceful degradation for existing applications: This is a requirement for some larger applications, especially when running on older devices where updates aren't always possible - that there would be an easy way to polyfill this proposal. While this isn't a gating requirement, using a secondary memory does make this harder.
 
-Challenges of using a single linear memory:
+## Proposed changes
 
- - Significant API changes to decouple the current 1:1 mapping for a WebAssembly.Memory object to a JS Memory object.
- - Potential performance degradation to default memory (however, the goal would be to minimize any effects to the default memory)
- - Engine implementations are more involved.
+### Core spec
+
+This proposal would add a `mappable` mode to WebAssembly memories:
```

Comment:
Per our discussion yesterday, I'm keeping these in the doc for now since they seem like a straightforward application of anonymous mappings in line with what you proposed for the JS API. We can adapt later based on other discussions.